### PR TITLE
Include cascading properties in Navigation deprecation

### PR DESCRIPTION
**Pull request by tellthemachines**
## Description

The Navigation block migration that handles the change to layout isn't adding `setCascadingProperties:"true"` to the attributes, so migrated navigations are not getting the correct justification properties set. 

This PR adds that attribute and cleans up the code a little.

## How has this been tested?

I think the easiest way to test this is switching to code view in the block editor, pasting in some deprecated nav markup such as `<!-- wp:navigation {"itemsJustification":"right","isResponsive":true} -->` and saving. The saved content should show the updated attributes, including `setCascadingProperties`. 

Moving out of code view and adding a few items to the nav, the layout should reflect the justification/orientation attributes set.

## Types of changes

Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Comment by jasmussen** [comment]
Code looks good to me, but I would think it might need a code sanity check beyond what I can offer. Thanks for working on this.

**Review by Mamaduka — APPROVED** [review]
It works as expected, and the code looks good to me. Thanks, @tellthemachines.

File: test/integration/fixtures/blocks/core__navigation__deprecated.json [hunk 1]
```diff
@@ -15,7 +15,7 @@
 			},
 			"layout": {
 				"type": "flex",
-				"justifyContent": "left",
+				"setCascadingProperties": "true",
```
Comment (youknowriad) [inline test/integration/fixtures/blocks/core__navigation__deprecated.json:18]: What does this property on a "layout" object mean? Sounds very weird to me?

Comment (youknowriad) [reply test/integration/fixtures/blocks/core__navigation__deprecated.json:18]: I see it's coming from #36292 I'll check there.

**Pull request merged**